sql/src/main/java/org/apache/druid/sql/calcite/schema/SystemSchema.java

```diff
@@ -755,6 +770,170 @@ public void close()
+        {
+          this.it = getAuthorizedSupervisors(tasks, root);
+        }
+
+        @Override
+        public Iterator<Object[]> iterator()
+        {
+          throw new UnsupportedOperationException("Do not use iterator(), it cannot be closed.");
+        }
+
+        @Override
+        public Enumerator<Object[]> enumerator()
+        {
+          return new Enumerator<Object[]>()
+          {
+            @Override
+            public Object[] current()
+            {
+              final SupervisorStatus supervisor = it.next();
+              return new Object[]{
+                  supervisor.getId(),
+                  supervisor.getState(),
+                  supervisor.getDetailedState(),
+                  supervisor.isHealthy() ? 1L : 0L,
+                  supervisor.getType(),
+                  supervisor.getSource(),
+                  supervisor.isSuspended() ? 1L : 0L,
+                  supervisor.getSpecString()
+              };
+            }
+
+            @Override
+            public boolean moveNext()
+            {
+              return it.hasNext();
+            }
+
+            @Override
+            public void reset()
+            {
+
+            }
+
+            @Override
+            public void close()
+            {
+              try {
+                it.close();
+              }
+              catch (IOException e) {
+                throw new RuntimeException(e);
+              }
+            }
+          };
+        }
+      }
+
+      return new SupervisorsEnumerable(getSupervisors(druidLeaderClient, jsonMapper, responseHandler));
+    }
+
+    private CloseableIterator<SupervisorStatus> getAuthorizedSupervisors(
+        JsonParserIterator<SupervisorStatus> it,
+        DataContext root
+    )
+    {
+      final AuthenticationResult authenticationResult =
+          (AuthenticationResult) root.get(PlannerContext.DATA_CTX_AUTHENTICATION_RESULT);
+
+      Function<SupervisorStatus, Iterable<ResourceAction>> raGenerator = supervisor -> Collections.singletonList(
+          AuthorizationUtils.DATASOURCE_READ_RA_GENERATOR.apply(supervisor.getSource()));
+
+      final Iterable<SupervisorStatus> authorizedTasks = AuthorizationUtils.filterAuthorizedResources(
+          authenticationResult,
+          () -> it,
+          raGenerator,
+          authorizerMapper
+      );
+
+      return wrap(authorizedTasks.iterator(), it);
+    }
+  }
+
+  // Note that overlord must be up to get supervisor tasks, otherwise queries to sys.supervisors table
+  // will fail with internal server error (HTTP 500)
+  private static JsonParserIterator<SupervisorStatus> getSupervisors(
+      DruidLeaderClient indexingServiceClient,
+      ObjectMapper jsonMapper,
+      BytesAccumulatingResponseHandler responseHandler
+  )
+  {
+    Request request;
+    try {
+      request = indexingServiceClient.makeRequest(
+          HttpMethod.GET,
+          StringUtils.format("/druid/indexer/v1/supervisor?fullStatus"),
```

Comment:
Is there a benefit of using `StringUtils.format()` if there are no args to be formatted?

Reply:
not sure, may be not, but i saw some places using this, and because of habit by now.